experiment/simulation/pages/factorial.html

```diff
@@ -0,0 +1,242 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Recursion | VLabs</title>
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/virtual-labs/virtual-style@0.0.8-b/css/style.min.css">
+    <link rel="stylesheet" type="text/css" href="../css/main.css" />
+</head>
+
+<body>
+    <div class="mdl-layout mdl-js-layout mdl-layout--fixed-drawer">
+        <div class="mdl-layout__drawer">
+            <span class="mdl-layout-title"><a href="../index.html" style="color:black;">Problems</a></span>
+            <nav class="mdl-navigation">
+                <a class="mdl-navigation__link" href="./add_numbers.html">Sum of Numbers</a>
+                <a class="mdl-navigation__link" href="./power.html">Power of a Number</a>
+                <!-- <a class="mdl-navigation__link" href="./towerofhanoi.html">Visualize Towers of Hanoi</a> -->
+                <a class="mdl-navigation__link nav-active" href="./factorial.html">Factorial of Number</a>
+                <a class="mdl-navigation__link" href="./fibonacci.html">Fibonacci series</a>
+            </nav>
+        </div>
+        <main class="mdl-layout__content">
+            <div class="demo-separator"></div>
+            <div class="mdl-grid">
+                <div class="mdl-cell mdl-cell--8-col mdl-cell--8-col-tablet mdl-cell--4-col-phone">
+                    <div class="v-collapsible-instruction-container v-instruction-container">
+                        <div class="v-collapsible is-size-5" id="task-description">
+                            Instructions
+                        </div>
+                        <div class="v-content" style="text-align: left; padding-left: 10%; padding-right: 10%;">
+                            <ul style="list-style: disc;">
+                                <li>Select the most appropriate option from the dropdowns in the code.</li>
+                                <li>Click on the <b>Run</b> button to run the code, and directly get the final output.</li>
+                                <li>Click on the <b>Next</b> button to see the next step of the code.</li>
+                                <li>Click on the <b>Prev</b> button to see the previous step of the code.</li>
+                                <li>Click on the <b>Reset</b> button to reset the code execution.</li>
+                            </ul>
+                        </div>
+                    </div>
+                </div>
+                <div class="mdl-cell mdl-cell--4-col mdl-cell--4-col-tablet mdl-cell--4-col-phone">
+                    <select id="lang-selector" class="lang-selector is-size-5">
+                        <option value="" disabled selected>Select a language</option>
+                        <option value="C">C</option>
```

Comment:
Similarly, adding a new language should be as easy as going to a config file and adding another entry. Here the list of languages is hard coded and it is defined once in each JS file, which makes for multiple sources of truth. Over time it becomes difficult to keep them in sync

Reply:
For adding a new language, it will be necessary to add a whole block of code in the HTML format, for which I am not sure if it would be possible to add in a markdown file and then include it. (As it needs to have information about the select options, which remain the same across all language examples.)

Reply:
The configuration will be in json format and not in md. You can read json and dynamically fill the values in the HTML. However, I do see some issues to be resolved here. Let's talk on the call today